src/components/TagContainer.ts

```diff
@@ -0,0 +1,237 @@
+        this.showErrorMessage = this.showErrorMessage.bind(this);
+
+        this.tagAttribute = props.tagAttribute.split("/")[props.tagAttribute.split("/").length - 1];
+        this.tagEntity = props.tagEntity.split("/")[props.tagEntity.split("/").length - 1];
+        this.referenceAttribute = props.tagEntity.split("/")[0]; // change to .length -1 instead of [0]
+    }
+
+    render() {
+        return createElement(Tag, {
+            alertMessage: this.state.alertMessage,
+            tagStyle: this.props.tagStyle,
+            className: this.props.class,
+            createTag: this.createTag,
+            enableSuggestions: this.props.enableSuggestions,
+            inputPlaceholder: this.props.inputPlaceholder,
+            lazyLoad: this.props.lazyLoad,
+            lazyLoadTags: this.lazyLoadTags,
+            newTag: this.state.newTag,
+            onRemove: this.removeTag,
+            readOnly: this.isReadOnly(),
+            showError: this.showErrorMessage,
+            style: parseStyle(this.props.style),
+            suggestions: this.state.suggestions,
+            tagLimit: this.props.tagLimit,
+            tagLimitMessage: this.props.tagLimitMessage,
+            tagList: this.state.tagList
+        });
+    }
+
+    componentWillReceiveProps(newProps: TagContainerProps) {
+        if (!this.props.lazyLoad) {
+            this.fetchCurrentTags(newProps.mxObject);
+        }
+        this.resetSubscriptions(newProps.mxObject);
+    }
+
+    componentWillUnmount() {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+    }
+
+    private showErrorMessage(message: string) {
+        window.mx.ui.error(message);
+    }
+
+    private getValue(attribute: string, mxObject?: mendix.lib.MxObject): string {
+        return mxObject ? (mxObject.get(attribute) as string) : "";
+    }
+
+    private isReadOnly() {
+        const { editable, mxObject, readOnly } = this.props;
+        if (editable === "default" && mxObject) {
+            return readOnly || mxObject.isReadonlyAttr(this.tagAttribute);
+        }
+
+        return true;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+
+            this.subscriptionHandles.push(mx.data.subscribe({
+                attr: this.tagAttribute,
+                callback: this.handleSubscriptions,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscriptions() {
+        this.setState({
+            newTag: this.getValue(this.tagAttribute, this.props.mxObject) as string
+        });
+    }
+
+    private lazyLoadTags() {
+        if (this.props.mxObject) {
+            this.fetchCurrentTags(this.props.mxObject);
```

Comment:
?? fetchCurrentTags ? when you are looking for the lazy load sugestion, I am lost

Reply:
Renamed